components/cost-management/policies/.chainsaw-test/chainsaw-test.yaml

```
@@ -0,0 +1,136 @@
+    in a tenant namespace
+  steps:
+    - name: Create namespaces for testing
+      try:
+        - create:
+            file: ./resources/namespace-cost-center.yaml
+            template: true
+            bindings:
+              - name: namespace
+                value: demo-tenant
+    - name: Apply RBAC
+      try:
+        - apply:
+            file: ../kyverno_rbac.yaml
+    - name: Apply kyverno Cluster Policy and assert it exists
+      try:
+        - apply:
+            file: ../update_namespace_labels.yaml
+        - assert:
+            file: chainsaw-assert-clusterpolicy.yaml
+    - name: create pods in demo-tenant
+      try:
+        - create:
+            file: ./resources/pod.yaml
+            bindings:
+              - name: namespace
+                value: demo-tenant
+            template: true
+    - name: assert pods in the demo-tenant are labeled
+      try:
+        - assert:
+            file: ./resources/expected-pod-matching.yaml
+            template: true
+            bindings:
+              - name: namespace
+                value: demo-tenant
+              - name: cost_center
+                value: team
+
+---
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: label-not-applied-random-ns
+spec:
+  concurrent: true
+  description: |
+    tests that the label is not applied to pods in a namespace
+    that does not match the label selector
+    in a tenant namespace
+  steps:
+    - name: Create namespaces for testing
+      try:
+        - create:
+            file: ./resources/namespace-nonmatching.yaml
+    - name: Apply kyverno Cluster Policy and assert it exists
+      try:
+        - apply:
+            file: ../update_namespace_labels.yaml
+        - assert:
+            file: chainsaw-assert-clusterpolicy.yaml
+    - name: create pods in random-ns
+      try:
+        - create:
+            file: ./resources/pod.yaml
+            bindings:
+              - name: namespace
+                value: random-ns
+            template: true
+    - name: assert pods in random-ns are not labeled
+      try:
+        - assert:
+            file: ./resources/pod.yaml
+            bindings:
+              - name: namespace
+                value: random-ns
+            template: true
+
+---
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: label-no-cost-center
+spec:
+  concurrent: true
+  description: |
+    tests that the label is applied to pods in a namespace
+    that matches the label selector
+    in a tenant namespace
+    and the cost_center label is set
+  steps:
+    - name: Create namespaces for testing
+      try:
+        - create:
+            file: ./resources/namespace-no-cost-center.yaml
+            template: true
+            bindings:
+              - name: namespace
+                value: demo-tenant-no-cost-center
+    # - name: Apply permission on namespaces
+    #   try:
+    #     - apply:
+    #         file: ../kyverno_rbac.yaml
```

this shouldn't be disabled

```suggestion
     - name: Apply permission on namespaces
       try:
         - apply:
             file: ../kyverno_rbac.yaml
```